src/pages/App.tsx

```diff
@@ -8,15 +8,29 @@ import { ToolBar } from '../components/ToolBar';
 import { RequestCards } from '../components/RequestCards';
 import { data } from 'data';
 import { ProductRequest } from 'types';
-
+import { NoFeedback } from 'components/NoFeedback';
 const App = () => {
   const [allData, setAllData] = useState(data);
   const [showData, setShowData] = useState<ProductRequest[]>(
     allData.productRequests,
   );
 
-  const { productRequests } = allData;
+  const filter = (productCategory: string) => {
+    let filteredItems = productRequests.filter((request) => {
+      return request.category === productCategory;
+    });
+    setShowData(filteredItems);
+  };
 
+  const showCards =
```

Comment:
I would avoid setting up a constant here, there's no reason that this code cannot live directly in line with the JSX return